compiler/rustc_lint/src/drop_forget_useless.rs

```diff
@@ -162,3 +202,10 @@ fn is_single_call_in_arm<'tcx>(
     }
     false
 }
+
+fn is_type_manually_drop(cx: &LateContext<'_>, ty: Ty<'_>) -> bool {
+    match ty.kind() {
+        ty::Adt(adt, _) => cx.tcx.lang_items().get(LangItem::ManuallyDrop) == Some(adt.did()),
```

Comment:
We already have that info stored in the ADT flags: https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/adt/struct.AdtDef.html#method.is_manually_drop